benchmarks/gateway-comparison/run-benchmark.sh

```
@@ -0,0 +1,290 @@
+    done
+    err "$name did not start within ${timeout}s"
+    return 1
+}
+
+# Resource monitor: samples RSS (KB) and CPU% every 0.5s
+start_monitor() {
+    local pid=$1 outfile=$2
+    (
+        echo "timestamp_ms,rss_kb,cpu_pct" > "$outfile"
+        while kill -0 "$pid" 2>/dev/null; do
+            read rss cpu <<< "$(ps -o rss=,pcpu= -p "$pid" 2>/dev/null || echo "0 0")"
```

Shellcheck (0.11.0)

[info] read without -r will mangle backslashes.

(SC2162)